feat(pie-docs): DSW-000 mention pie-webc more and remove next13 examples (#2089)
justeattakeaway/pie: apps/pie-docs/src/engineers/web-components/overview.md

Currently, this technology is supported by all browsers, making it possible to have one central library that will work with most frontend frameworks.

> The long-term aim will be to migrate JET teams over to this Web Component system, so we have one single source of truth for our global PIE components.

Reviewer: ```suggestion
The long-term aim is to migrate JET teams over to this Web Component system, so we have one single source of truth for our global PIE components.
```
